### PR TITLE
Add new fields to stg_serverless_task_history

### PR DESCRIPTION
**Pull request by fernandobrito**
Improve the support for Snowflake tasks.

### CODE REVIEW

File: models/staging/stg_task_versions.sql [hunk 1]
```diff
@@ -0,0 +1,35 @@
+{# Every task belong to a graph. First we identify the graph version and then the task id #}
```
Comment (ian-whitestone) [inline models/staging/stg_task_versions.sql:1]: should this comment be here?

File: models/cost_per_serverless_task_run.sql [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        and task_schema_id is not null
+        and task_database_id is not null
+        and startswith(warehouse_name, 'COMPUTE_SERVICE_WH_USER_TASKS') -- TODO: Verify on a larger sample
+        and end_time <= (select latest_ts from stop_threshold)
+        {% if is_incremental() %}
+        -- account for late arriving queries
+        and end_time > (select coalesce(dateadd(day, -3, max(end_time)), '1970-01-01') from {{ this }})
+        {% endif %}
+
+    order by scheduled_time desc
```
Comment (ian-whitestone) [inline models/cost_per_serverless_task_run.sql:37]: styling nitpick, but can you please prefix the columns with the table they came from, i.e.

```sql
and task_history.task_database_id is not null)
and startswith(query_history.warehouse_name, 'COMPUTE_SERVICE_WH_USER_TASKS')
...
order by task_history.scheduled_time desc
```

File: models/cost_per_serverless_task_run.sql [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+        query_history.*,
+        task_history.database_name as task_database_name,
+        task_history.schema_name as task_schema_name,
+        name as task_name,
+        concat(task_database_name, '.', task_schema_name, '.', task_name) as task_fqn_name,
+        start_time as execution_start_time,
+    from {{ ref('stg_task_history') }} as task_history
+
+    inner join {{ ref('stg_query_history') }} as query_history
+        on task_history.query_id = query_history.query_id and query_history.start_time = task_history.query_start_time
```
Comment (ian-whitestone) [inline models/cost_per_serverless_task_run.sql:25]: confirming that joining on `start_time=query_start_time` is safe?

File: models/cost_per_serverless_task_run.sql (hunk 2 above)
Comment (ian-whitestone) [reply models/cost_per_serverless_task_run.sql:37]: oh also, i don't htink we need this sort in here?

File: models/cost_per_serverless_task_run.sql [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+    select
+        start_time as hour,
+        database_name,
+        database_id,
+        schema_name,
+        schema_id,
+        task_name,
+        credits_used as credits_used_serverless_task
+    from {{ ref('stg_serverless_task_history') }}
+    group by all
```
Comment (ian-whitestone) [inline models/cost_per_serverless_task_run.sql:88]: should we do:

```sql
    select
        start_time as hour,
        database_id,
        schema_id,
        task_name,
        sum(credits_used) as credits_used_serverless_task
    from {{ ref('stg_serverless_task_history') }}
    group by all
```

to enforce the desired grain? we later join on `database_id, schema_name, task_id` so not sure why we need the other columns kicking around?

File: models/cost_per_serverless_task_run.sql [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+        any_value(task_database_name) as task_database_name,
+        any_value(task_schema_id) as task_schema_id,
+        any_value(task_schema_name) as task_schema_name,
+        any_value(task_name) as task_name,
+        any_value(execution_start_time) as execution_start_time,
+        sum(allocated_serverless_task_credits_in_hour) as serverless_task_credits,
+        sum(allocated_serverless_task_cost_in_hour) as serverless_task_cost,
+    from run_cost
+    group by 1
+)
```
Comment (ian-whitestone) [inline models/cost_per_serverless_task_run.sql:123]: if we dedupe `credits_billed_hourly` as per my comment above, is this aggregation here necessary?

File: models/cost_per_query.sql [hunk 6]
```diff
@@ -173,14 +205,16 @@ select
...
     -- from that day are not allocated any cloud_services_cost. The next time the model runs, after we have the full day of data,
     -- this may change if cloud credits make up >10% of compute cost.
     (div0(all_queries.credits_used_cloud_services, credits_billed_daily.daily_credits_used_cloud_services) * credits_billed_daily.daily_billable_cloud_services) * coalesce(daily_rates.effective_rate, current_rates.effective_rate) as cloud_services_cost,
     div0(all_queries.credits_used_cloud_services, credits_billed_daily.daily_credits_used_cloud_services) * credits_billed_daily.daily_billable_cloud_services as cloud_services_credits,
-    all_queries.compute_cost + all_queries.query_acceleration_cost + cloud_services_cost as query_cost,
-    all_queries.compute_credits + all_queries.query_acceleration_credits + cloud_services_credits as query_credits,
+    all_queries.compute_cost + all_queries.query_acceleration_cost + all_queries.serverless_task_cost + cloud_services_cost as query_cost,
+    all_queries.compute_credits + all_queries.query_acceleration_credits + all_queries.serverless_task_credits + cloud_services_credits as query_credits,
```
Comment (ian-whitestone) [inline models/cost_per_query.sql:217]: @NiallRees aligned on putting this into here?

File: models/cost_per_serverless_task_run.sql (hunk 4 above)
Comment (fernandobrito) [reply models/cost_per_serverless_task_run.sql:88]: Good idea. I added the database and schema names when I was troubleshooting it, but indeed we only use the IDs. I just removed the other fields, thanks!

File: models/cost_per_serverless_task_run.sql (hunk 2 above)
Comment (fernandobrito) [reply models/cost_per_serverless_task_run.sql:37]: I agree with the style comment. The `order by` was something I forgot while working on the query on Snowsigth and then moving here to the model. I just removed it, thanks!

File: models/cost_per_serverless_task_run.sql (hunk 3 above)
Comment (fernandobrito) [reply models/cost_per_serverless_task_run.sql]: Yes! I think there was a misunderstanding between you and Teghan. He meant something else with start_time not matching (parent vs. children), and not what I think you understood (Snowflake claiming that two columns in different tables should match and they don't).

File: models/staging/stg_task_versions.sql (hunk 1 above)
Comment (fernandobrito) [reply models/staging/stg_task_versions.sql]: Not really needed.

**Pull request merged**